### PR TITLE
CXP-1242: Rewrite internal controllers & add update of enabled

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

**Main change °1:**

Instead of doing

`POST /rest/catalogs/{catalogId}/save-criteria`
```
[
  [
    field: 'enabled',
    operator: '=',
    value: true
  ]
]
```

now, it's `PATCH /rest/catalogs/{catalogId}`
```
{
  enabled: true,
  product_selection_criteria: [
    [
      field: 'enabled',
      operator: '=',
      value: true
    ]
  ]
}
```

**Main change °2:**

`GET /rest/catalogs/{catalogId}` also returns `product_selection_criteria`

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQuery.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public function execute(string $id): ?array
         ])->fetchOne();
 
         if (!$result) {
-            return null;
+            throw new \LogicException('Catalog not found');
```
Comment (RiffVentura) [inline components/catalogs/back/src/Infrastructure/Persistence/FindCatalogProductSelectionCriteriaQuery.php:40]: rename file and interface to `GetCatalogProductSelectionCriteriaQuery`

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Controller/Internal/GetCatalogAction.php [hunk 2]
```diff
@@ -36,6 +38,14 @@ public function __invoke(Request $request, string $catalogId): Response
...
-        return new JsonResponse($this->normalizer->normalize($catalog, 'internal'));
+        $catalogNormalized = (array) $this->normalizer->normalize($catalog, 'internal');
+        $productSelectionCriteria = $this->findCatalogProductSelectionCriteriaQuery->execute($catalogId);
+
+        return new JsonResponse(\array_merge(
+            $catalogNormalized,
+            [
+                'product_selection_criteria' => $productSelectionCriteria,
+            ],
+        ));
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Controller/Internal/GetCatalogAction.php:49]:
```suggestion
        $catalogNormalized = (array) $this->normalizer->normalize($catalog, 'internal');
        $catalogNormalized['product_selection_criteria'] = $this->findCatalogProductSelectionCriteriaQuery->execute($catalogId);

        return new JsonResponse($catalogNormalized);
```
It's more readable.

File: components/catalogs/back/src/Infrastructure/Controller/Internal/UpdateCatalogAction.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                ],
+                Response::HTTP_UNPROCESSABLE_ENTITY
+            );
+        }
+
+        $this->upsertCatalogQuery->execute(
+            id: $catalogId,
+            name: $catalog->getName(),
+            ownerUsername: $catalog->getOwnerUsername(),
+            enabled: $payload['enabled'],
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Controller/Internal/UpdateCatalogAction.php:71]: any particular reason you use named parameters here?

**Review by masacc — APPROVED** [review]

**Review by willy-ahva — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Controller/Internal/UpdateCatalogAction.php (hunk 3 above)
Comment (masacc) [reply components/catalogs/back/src/Infrastructure/Controller/Internal/UpdateCatalogAction.php:71]: I won't say I don't like this syntax but we don't use it anywhere else. So we could decide when/where to use it all together if you want?

**Review by masacc — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Persistence/UpdateCatalogProductSelectionCriteriaQuery.php [hunk 4]
```diff
@@ -30,7 +30,7 @@ public function execute(string $id, array $productSelectionCriteria): void
             $query,
             [
                 'id' => Uuid::fromString($id)->getBytes(),
-                'product_selection_criteria' => $productSelectionCriteria,
+                'product_selection_criteria' => array_values($productSelectionCriteria),
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Persistence/UpdateCatalogProductSelectionCriteriaQuery.php:33]:
```suggestion
                'product_selection_criteria' => \array_values($productSelectionCriteria),
```

File: components/catalogs/back/src/Infrastructure/Validation/UpdateCatalogPayloadIsValid.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                                'allowMissingFields' => false,
+                                'allowExtraFields' => false,
+                            ]),
+                        ),
+                        new Assert\Callback(function ($array, ExecutionContextInterface $context) {
+                            if (!is_array($array)) {
+                                return;
+                            }
+
+                            if (count(array_filter(array_keys($array), 'is_string')) > 0) {
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Validation/UpdateCatalogPayloadIsValid.php:48]:
```suggestion
                            if (\count(\array_filter(\array_keys($array), 'is_string')) > 0) {
```

Comment (tseho) [reply components/catalogs/back/src/Infrastructure/Validation/UpdateCatalogPayloadIsValid.php:48]: I forgot to run the tests locally, the CI will not let me pass :)

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**